shared/chat/conversation/input-area/normal/mention-input.js

```diff
@@ -1,39 +1,49 @@
 // @flow
+/* eslint-disable sort-keys */
 import * as React from 'react'
 import PlatformInput from './platform-input'
 import {type MentionInputProps} from './types'
 import {Input} from '../../../../common-adapters'
 import {isMobile} from '../../../../constants/platform'
 
 type MentionState = {|
-  pickSelectedCounter: number,
-  mentionFilter: string,
   channelMentionFilter: string,
-  mentionPopupOpen: boolean,
   channelMentionPopupOpen: boolean,
+  mentionFilter: string,
+  mentionPopupOpen: boolean,
+  pickSelectedCounter: number,
 
   // Desktop only.
-  upArrowCounter: number,
   downArrowCounter: number,
+  upArrowCounter: number,
 |}
 
+type SelectionKey = 'Enter' | 'Tab' | ''
+
 class MentionInput extends React.Component<MentionInputProps, MentionState> {
   state: MentionState = {
-    upArrowCounter: 0,
+    channelMentionFilter: '',
+    channelMentionPopupOpen: false,
     downArrowCounter: 0,
-    pickSelectedCounter: 0,
     mentionFilter: '',
-    channelMentionFilter: '',
     mentionPopupOpen: false,
-    channelMentionPopupOpen: false,
+    pickSelectedCounter: 0,
+    upArrowCounter: 0,
   }
+
+  _lastSelectionKey: SelectionKey = ''
+
   _inputRef: ?Input
 
   _inputSetRef = (input: ?Input) => {
     this.props.inputSetRef(input)
     this._inputRef = input
   }
 
+  _setLastSelectionKey = (lastSelectionKey: SelectionKey) => {
+    if (this._lastSelectionKey !== lastSelectionKey) this._lastSelectionKey = lastSelectionKey
```

Comment:
it's probably not worth checking before setting, so just `this._lastSelectionKey = lastSelectionKey`